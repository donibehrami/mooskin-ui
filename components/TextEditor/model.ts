export interface ITextEditorComponentProps {

    /** id assigned to the editor */
    editorId?: string;

    /** selects different element to attach the editor to */
    selector?: any;

    /** additional plugins to be added to the RTE */
    plugins?: string[];

    /** show/hide the menu bar */
    menubar?: boolean;

    /** makes the editor read only */
    disabled?: boolean;

    /** makes the toolbar and menu bar inline, pops up on text area focus */
    inline?: boolean;

    /** what options should be available in the toolbar */
    toolbar?: string;

    /** initial value of the editor (HTML string) */
    initialValue?: string;

    /** if the return format should be html or text */
    outputFormat?: 'html' | 'text';

    /** value of the editor */
    value?: string;

    /** Only valid when <Editor inline={true} />. Used to define the HTML element for the editor in inline mode. */
    tagName?: string;

    /** Sets the name attribute for the textarea element used for the editor in forms. */
    textareaName?: string;

    /** initialization object */
    init?: {[key: string]: any};

    /** onChange event handler when changing the editor */
    onEditorChange?: (content: string, editor: any) => void;
}
