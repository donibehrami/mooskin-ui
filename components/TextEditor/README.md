# MooSkin UI - Table Component

The MooSkin Button Component works similar to the normal HTML `<table/>` element, but with pre-defined styling and attributes.

___

### Usage

To start using the Table Component first you have to Import it

```
Import {Table, TableHeader} from '@moosend/mooskin';
```
or modular import
```
// JS
import Table, {TableHeader} from '@moosend/mooskin/lib/Table';

// CSS
import '@moosend/mooskin/lib/Table/style.css';
```

Styles can also be imported this way, if they were not imported globaly and you want to import just the table styles to keep the css bundle size healthy: 

```
import '@moosend/mooskin/lib/Table/styles.css';
```

And then you can simply start using it by typing

```
const data = [
    {
        name: 'Geralt of Rivia',
        weapon: 'Aerondight',
        abilities: 'Sword fighting, Quen Signs',
        profession: 'Witcher',
        allegience: 'School of the Wolf'
    },
    {
        name: 'John Shepard',
        weapon: 'Vindicator',
        abilities: 'Weapons, Biotics',
        profession: 'Commander',
        allegience: 'Alliance, Citadel'
    }
];

<Table data={data}>
    <TableHeader dataField="name" >Name</TableHeader>
    <TableHeader dataField="profession" hideSmall >Profession</TableHeader>
    <TableHeader dataField="weapon" hideSmall >Weapon</TableHeader>
    <TableHeader dataField="abilities" >Abilities</TableHeader>
    <TableHeader dataField="allegience" hideSmall >Allegience</TableHeader>
</Table>
```
Basically the table accepts an array of objects as a prop, those objects it will assign to rows and columns. One object is a row and an object property is a column. Table accepts `TableHeader` components and children, where you can assign which property to which column should be assigned. For example the `abilities` property from the objects will be assigned to the 4th column of the table determined by `TableHeader` component and the `dataField` prop.

Table component can recieve many props. The `sortable` prop for example
```
<Table data={data}>
    <TableHeader dataField="property1" sortable >Header 1</TableHeader>
    <TableHeader dataField="property2" >Header 2</TableHeader>
    <TableHeader dataField="property3" >Header 3</TableHeader>
</Table>
```
This will make the first column sortable by clicking on the header.

A custom sort function can be passed aswell to headers.

```
<Table data={data}>
    <TableHeader dataField="property1" sortable sortfn={this.sortIt} >Header 1</TableHeader>
    <TableHeader dataField="property2" >Header 2</TableHeader>
    <TableHeader dataField="property3" >Header 3</TableHeader>
</Table>

sortIt(a, b, order, sortBy) {

    let comparison = 0;

    if (order === 'desc'){

        console.log('desc called');

        if (a[sortBy] > b[sortBy]){
            comparison = -1;
        } else if (a[sortBy] < b[sortBy]){
            comparison = 1;
        }
        return comparison;
    } else if (order === 'asc') {

        console.log('asc called');

        if (a[sortBy] < b[sortBy]){
            comparison = -1;
        } else if (a[sortBy] > b[sortBy]){
            comparison = 1;
        }
        return comparison;
    }

}
```
The custom sort function must accept 4 arguments, `a` and `b` being the compared objects, `order` must bo `asc` or `desc` and `sortBy` is the object property to be compared.

The table can also be paginated using the `paginate` props,

```
<Table data={data} paginate={3}>
    <TableHeader dataField="property1" sortable sortfn={this.sortIt} >Header 1</TableHeader>
    <TableHeader dataField="property2" >Header 2</TableHeader>
    <TableHeader dataField="property3" >Header 3</TableHeader>
</Table>
```
in this case the table will be paginated, showing only three rows since a value of 3 is passed, and pagination buttons will appear.

### Row styles via data

Specific rows can be styled by passing a `style` key through the object in the `data` prop.

```
const data = [
    {
        name: 'Geralt of Rivia',
        weapon: 'Aerondight',
        abilities: 'Sword fighting, Quen Signs',
        profession: 'Witcher',
        allegience: 'School of the Wolf'
    },
    {
        name: 'John Shepard',
        weapon: 'Vindicator',
        abilities: 'Weapons, Biotics',
        profession: 'Commander',
        allegience: 'Alliance, Citadel',
        style: {color: '#5ccdde'}
    }
];

<Table data={data}>
    <TableHeader dataField="name" >Name</TableHeader>
    <TableHeader dataField="profession" hideSmall >Profession</TableHeader>
    <TableHeader dataField="weapon" hideSmall >Weapon</TableHeader>
    <TableHeader dataField="abilities" >Abilities</TableHeader>
    <TableHeader dataField="allegience" hideSmall >Allegience</TableHeader>
</Table>
```

in this case, the second object in the array has a style key value, that style will be applied to the second row of the Table, more precicly to all cells in that specific row.

### Row (Cell) callback functions via data

Specific rows can have callback functions by passing a `onClick` key through the object in the `data` prop.

```
const data = [
    {
        name: 'Geralt of Rivia',
        weapon: 'Aerondight',
        abilities: 'Sword fighting, Quen Signs',
        profession: 'Witcher',
        allegience: 'School of the Wolf'
    },
    {
        name: 'John Shepard',
        weapon: 'Vindicator',
        abilities: 'Weapons, Biotics',
        profession: 'Commander',
        allegience: 'Alliance, Citadel',
        onClick: (e, data) => console.log(e, data);
    }
];

<Table data={data}>
    <TableHeader dataField="name" >Name</TableHeader>
    <TableHeader dataField="profession" hideSmall >Profession</TableHeader>
    <TableHeader dataField="weapon" hideSmall >Weapon</TableHeader>
    <TableHeader dataField="abilities" >Abilities</TableHeader>
    <TableHeader dataField="allegience" hideSmall >Allegience</TableHeader>
</Table>
```

the callback will return two args, `e` for event & `data` which is an object consisting of the relevant `dataLabel` of the cell, `label` which is the text, or content, of the clicked cell, `index` is the index of the row in the data array & `content` which is the inner content of the cell

### Expanded sections for Rows

As each row is represented by a object in the array that is passed to the `data` prop, an expanded section can be added to the Row, similar to the [List](https://github.com/moosend/mooskin-ui/tree/master/components/List) component. The expanded section will be rendered by the `expandable` property of the Rows object. The `expandable` property should be an object, consisting of two mandatory properties (`expanded`: boolean, `content`: a TD JSX element, or an array of such elements. When rendered, the props, key & children of the element/s will be the same, though the first cell will have the style of `position: 'relative'` passed to it, since its required to render the absolute section arrow, also on the first cell, apart from its children, the arrow will be added as a child. 'colSpan' table cell prop should be used to navigate/style cells in the row. The optional ones being (`className`: string, `style`: CSS properties). The proper way to change the `expandable` section properties, is by using the above mentioned `onClick` callbacks for Rows.

Example:
```
const data = [
    {
        name: 'Geralt of Rivia',
        weapon: 'Aerondight',
        abilities: 'Sword fighting, Quen Signs',
        profession: 'Witcher',
        allegience: 'School of the Wolf',
        expandable: {
            expanded: true,
            content: <div style={{background: 'red'}}>This section is expandable</div>
        }
    },
    {
        name: 'John Shepard',
        weapon: 'Vindicator',
        abilities: 'Weapons, Biotics',
        profession: 'Commander',
        allegience: 'Alliance, Citadel',
        onClick: (e, data) => console.log(e, data);
    }
];

<Table data={data}>
    <TableHeader dataField="name" >Name</TableHeader>
    <TableHeader dataField="profession" hideSmall >Profession</TableHeader>
    <TableHeader dataField="weapon" hideSmall >Weapon</TableHeader>
    <TableHeader dataField="abilities" >Abilities</TableHeader>
    <TableHeader dataField="allegience" hideSmall >Allegience</TableHeader>
</Table>
```


<div class="playground-doc">

## Supported attributes for Table

* `id` - id of the element
* `data` - array of objects to be assigned to the Table
* `rowStyles` - styles to be applied on the table Rows
* `rowClasses` - classes to be applied on the table Rows
* `className` - css class
* `style` - table styles
* `alternate` - alternate table view
* `smallCollapse` - adds another header and column on small screens where the whole row can be collapsed
* `collapseHeaderClassName` - classes for popover header
* `collapseHeaderStyle` - styles for popover header
* `containerStyle` - styling applied to the div containing the table
* `paginate` - paginate table, with the number of rows passed to this prop
* `paginationProps` - an object describing table pagination props
* `dragAndDrop` - an object that enables drag and drop functionality and includes position property in which column the drag icon should show, a function that tracks hover information and returns dragIndex and hoverIndex values while dragging a table row, and a drop function which will be called once we drop the row being dragged

## Supported attributes for TableHeader

* `dataField` - which object property of the data array should be assigned to this column
* `hideSmall` - Hides the column on small screens
* `colClasses` - classes to be applied to this column cells
* `colStyles` - styles to be applied to this column cells
* `sortable` - wether this column should be sortable or not
* `sortfn` - custom sort function for row sorting, requires `sortable`
* `arrowDirection` - determines the direction of the sorting arrow
* `className` - css class
* `style` - table styles

Allthough these attributes are supported, only `data` for Table and `dataField` for TableHeader are required.

</div>

#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)
