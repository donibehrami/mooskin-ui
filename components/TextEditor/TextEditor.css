@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';

.editor{
    outline: none;
    padding: 0 15px;
    font-family: Arial,sans-serif;
    border: 0;
    /* color: var(--inputFontColor); */
    /* font-size: 1rem; */
    border-radius: 8px;
    background: #fff;
    width: 100%;
    transition: 0.2s;
    border: 2px solid #efefef;
    max-height: 400px;
    overflow: auto;
    /* max-width: 100%; */
}

.textArea{
    composes: editor;
    height: 400px;
    overflow: auto;
    white-space: nowrap;
}

.buttonGroup{
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
}

.wrapper{
    display: flex;
    flex-direction: column;
}

.wrapperReverse{
    display: flex;
    flex-direction: column-reverse;
}

.option{
    border: none;
    padding: 5px;
    min-width: 25px;
    height: 20px;
    border-radius: 2px;
    margin: 0 4px;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    background: white;
    text-transform: capitalize;
}

.emojiOption{
    display: flex;
    flex-direction: row;
}

.option:hover {
    box-shadow: inset 0 0 2px #bfbfbf;
}

.dropDown {
    border: none;
}

/* .option img {
    width: 14px;
}

.emojiOption > img {
    width: 14px;
}

.group > img {
    width: 14px;
}

.group > a > img {
    width: 14px;
}

.customContainer > img {
    width: 16px;
}

:global(.rdw-history-wrapper img){
    width: 12px;
}

:global(.rdw-image-wrapper img){
    width: 17px;
} */

/* .group { */
    /* border-left: 1px solid #e5e5e5; */
    /* border-right: 1px solid #e5e5e5;
} */

/* .groupRight {
    border-right: 1px solid #e5e5e5;
}

.groupLeft {
    border-left: 1px solid #e5e5e5;
}

:global(.rdw-colorpicker-wrapper){
    border-right: 1px solid #e5e5e5;
}

:global(.rdw-dropdown-optionwrapper){
    left: -10px;
} */

/* :global(.rdw-colorpicker-wrapper){
    border-right: 2px solid rgba(241, 241, 241, 0.829);
}

:global(.rdw-image-wrapper){
    border-right: 2px solid rgba(241, 241, 241, 0.829);
} */

:global(.rdw-link-decorator-icon){
    left: 100%;
    content:url("../../assets/images/editor/GoTolink.png");
}

:global(.rdw-image-alignment-options-popup){
    height: 37px;
}

:global(.rdw-image-alignment-option){
    padding: 12px !important;
}

:global(.rdw-link-modal){
    display: flex;
    height: 240px;
}

:global(.rdw-dropdown-wrapper){
    margin: 0px;
    padding: 0px 5px;
}

:global(.rdw-dropdown-selectedtext){
    padding: 0;
    color: #444444;
}

/* .toolbar > .group:not(:last-child) > .option:last-child {
    margin-right: 11px;
}

.toolbar > .group:not(:first-child) > .option:first-child {
    margin-left: 11px;
} */

/* .editor::-webkit-scrollbar {
    width: 5px;
    height: 10px;
} */

/* .editor::-webkit-scrollbar-button:start:decrement,
.editor::-webkit-scrollbar-button:end:increment  {
    display: none;
} */

/* .editor::-webkit-scrollbar-track-piece  {
    background-color: #eeeeee;
} */

/* editor::-webkit-scrollbar-thumb:vertical {
    background: #d9d9d9;
    border-radius: 16px;
} */

.editorContainer{
    position: relative;
}

.overlay{
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    z-index: 2;
}

.editorLabel{
    color: var(--inputLabel);
    /* font-size: 13px;*/
    padding-left: 5px;
    font-size: .92857142857rem;
}

.toolbar{
    border-radius: 4px;
    border: 1px solid #efefef;
    /* flex-wrap: nowrap; */
    margin-top: 10px;
    color: #4c4c4c;
}

.toolbarOnFocus {
    position: absolute;
    z-index: 5;
    max-width: 700px;
}

.draggable {
    cursor: move;
}

:global(.mooskin-text-editor .rdw-colorpicker-modal){
    width: 300px;
    height: 160px;
}

:global(.mooskin-text-editor .rdw-colorpicker-modal-options){
    overflow-x: hidden;
    overflow-y: hidden;
    width: 100%;
}

:global(.rdw-link-modal-input) {
    width: 100%;
    padding: 5px;
}

:global(.rdw-link-modal-target-option){
    margin-bottom: 10px;
}

:global(.rdw-link-modal-target-option > input[type='checkbox']){
    width: 13px;
    height: 13px;
}

:global(.rdw-dropdown-carettoopen){
    top: 40%;
    border-top: 5px solid #4c4c4c;
    border-left: 3px solid transparent;
    border-right: 3px solid transparent;
}

:global(.rdw-dropdown-carettoclose){
    top: 40%;
    border-bottom: 5px solid #4c4c4c;
    border-left: 3px solid transparent;
    border-right: 3px solid transparent;
}

:global(.rdw-fontsize-dropdown){
    padding: 0px 11px;
    width: 60px;
    margin: 0;
}

:global(.rdw-option-wrapper){
    border: none;
}

/* :global(.public-DraftStyleDefault-block){
    white-space: nowrap;
} */

/* :global(.rdw-colorpicker-wrapper){
    padding: 0px 11px;
} */

/* :global(.rdw-emoji-wrapper){
    padding: 0px 5px 0px 5px;
    border-right: 1px solid #e5e5e5;
}

:global(.rdw-image-wrapper){
    padding: 0px 8px;
    border-right: 1px solid #e5e5e5;
}

:global(.rdw-remove-wrapper){
    padding: 0px 8px;
    border-right: 1px solid #e5e5e5;
} */

/* .editorContainer .rdw-link-modal{
    height: 220px;
} */

.alignDropdown {
    width: 48px;
    overflow-y: unset;
}

.customContainer{
    position: relative;
    display: flex;
    align-self: center;
    padding: 5px 0;
    min-width: 25px;
    width: 50px;
    height: 30px;
    border-radius: 2px;
    margin-bottom: 6px;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    background: white;
    text-transform: capitalize;
    position: relative;
}

.tagsContainer {
    composes: customContainer;
}

.tagsLabel {
    align-self: center;
    margin: 0 4px;
    /* display: flex; */
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

.arrowDown {
    height: 0px;
    width: 0px;
    /* position: absolute; */
    margin-left: 5px;
    border-top: 5px solid #4c4c4c;
    border-left: 3px solid transparent;
    border-right: 3px solid transparent;
}

.arrowUp {
    height: 0px;
    width: 0px;
    /* position: absolute; */
    margin-left: 5px;
    border-bottom: 5px solid #4c4c4c;
    border-left: 3px solid transparent;
    border-right: 3px solid transparent;
}

.tagsList{
    position: absolute;
    left: 0px;
    right: 0px;
    top: 30px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    background: #fff;
    border: 1px solid #F1F1F1;
    z-index: 10;
    width: 180px;
    max-height: 162px;
    overflow-y: auto;
}

.tag {
    padding: 5px;
    width: 100%;
}

.tag:hover {
    background: #F1F1F1;
}

.disabled {
    display: none;
}

.htmlContainer {
    position: relative;
    align-self: center;
    user-select: contain !important;
}

.htmlContent {
    position: absolute;
    left: 4px;
    top: 20px;
    background: #fff;
    border: 1px solid #F1F1F1;
    z-index: 6;
    padding: 20px;
    /* overflow: auto; */
    height: 500px;
    width: 500px;
    user-select: contain !important;
}

.copyHtml{
    position: absolute;
    top: 20px;
    left: 4px;
    background: rgb(173, 173, 173);
    font-size: 12px;
    width: 68px;
    z-index: 7;
    cursor: pointer;
}

.separator {
    border-left: 1px solid #F1F1F1;
    height: 30px;
    padding-left: 5px;
    margin-left: 5px;
}

:global(.public-DraftStyleDefault-ol){
    padding-left: 45px;
    margin: 30px 0;
}

:global(.public-DraftStyleDefault-ul){
    padding-left: 40px;
    margin: 30px 0;
}

:global(.public-DraftStyleDefault-depth0.public-DraftStyleDefault-listLTR){
    margin: -14px 0;
}
/* @media (--xSmall){

    :global(.mooskin-textarea-editor .rdw-colorpicker-modal){
        position: fixed;
        top: 25%;
        left: 15%;
    }

    :global(.mooskin-textarea-editor .rdw-link-modal){
        position: fixed;
        top: 25%;
        left: 20%;
    }

    :global(.mooskin-textarea-editor .rdw-embedded-modal){
        position: fixed;
        top: 25%;
        left: 20%;
    }

    :global(.mooskin-textarea-editor .rdw-emoji-modal){
        position: fixed;
        top: 25%;
        left: 20%;
    }

    :global(.mooskin-textarea-editor .rdw-image-modal){
        position: fixed;
        top: 25%;
        left: 20%;
    }

} */


@media (--deviceXsmall){
    .toolbar{
        top: -170px;
    }
}