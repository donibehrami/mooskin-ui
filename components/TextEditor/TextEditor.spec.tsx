import * as React from 'react';
import TextEditor from './TextEditor';

import { Editor } from 'react-draft-wysiwyg';

import { mount, shallow } from 'enzyme';

describe('TextEditor', () => {

    // test('renders correctly', () => {
    //     const func = jest.fn();

    //     const tree = shallow(
    //         <TextEditor
    //             label="label"
    //             editorClassName="myClass"
    //             editorStyle={{color: 'blue'}}
    //             onChange={func}
    //             toolbarClassName="toolbarClass"
    //             wrapperClassName="wrapperClass"
    //             draggable
    //             toolbarOnFocus
    //             wrapperStyle={{color: 'red'}}
    //             toolbarStyle={{background: 'green'}}
    //         />
    //     );
    //     expect(tree).toMatchSnapshot();
    // });

    test('renders correctly with custom editor props', () => {
        const func = jest.fn();

        const component = shallow(
            <TextEditor
                label="label"
                editorClassName="myClass"
                editorStyle={{color: 'blue'}}
                onChange={func}
                toolbarClassName="toolbarClass"
                wrapperClassName="wrapperClass"
            />
        );

        expect(component.find(Editor).prop('editorClassName')).toEqual('myClass');
        expect(component.find(Editor).prop('editorStyle')).toEqual({color: 'blue'});
        expect(component.find(Editor).prop('toolbarClassName')).toEqual('toolbarClass');

        component.find(Editor).simulate('change');

        expect(func).toHaveBeenCalled();
    });

    test('callback function is called on editor change', () => {
        const func = jest.fn();

        const component = shallow(
            <TextEditor
                onChange={func}
            />
        );

        component.find(Editor).simulate('change');

        expect(func).toHaveBeenCalled();
    });

    // test('position changes when editor is draggable and style classes are passed down', () => {
    //     const func = jest.fn();

    //     const component = mount(
    //         <TextEditor
    //             draggable
    //             toolbarClassName="customToolbar"
    //             wrapperClassName="customWrapper"
    //         />
    //     );

    //     expect(component.state('pos')).toEqual({left: 0, top: -70});

    //     expect(component.find('div').first().prop('className')).toContain('customWrapper');
    //     expect(component.find('div').at(2).prop('className')).toContain('customToolbar');

    //     component.find('div').at(2).simulate('mousedown');

    //     component.find('div').at(2).simulate('mousemove', {
    //         pageX: 42,
    //         pageY: 44
    //     });

    //     expect(component.state('pos')).toEqual({left: 0, top: -70});

    // });

});
