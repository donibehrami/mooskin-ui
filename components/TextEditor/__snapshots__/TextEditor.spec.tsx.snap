// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor renders correctly 1`] = `
<TextEditor
  disabled={false}
  init={Object {}}
  inline={false}
  menubar={false}
  onEditorChange={[MockFunction]}
  plugins={
    Array [
      "lists advlist",
      "anchor",
      "autolink",
      "autoresize",
      "charmap",
      "code codesample",
      "directionality",
      "emoticons",
      "fullscreen",
      "help",
      "hr",
      "image imagetools",
      "insertdatetime",
      "link",
      "nonbreaking",
      "paste",
      "preview",
      "print",
      "quickbars",
      "searchreplace",
      "table",
      "toc",
      "visualblocks",
      "wordcount",
    ]
  }
  toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
  value="<p>TextEditor</p>"
>
  <Editor
    apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
    cloudChannel="5"
    disabled={false}
    init={
      Object {
        "menubar": false,
        "selector": undefined,
        "setup": undefined,
      }
    }
    inline={false}
    menubar={false}
    onEditorChange={[MockFunction]}
    plugins={
      Array [
        "lists advlist",
        "anchor",
        "autolink",
        "autoresize",
        "charmap",
        "code codesample",
        "directionality",
        "emoticons",
        "fullscreen",
        "help",
        "hr",
        "image imagetools",
        "insertdatetime",
        "link",
        "nonbreaking",
        "paste",
        "preview",
        "print",
        "quickbars",
        "searchreplace",
        "table",
        "toc",
        "visualblocks",
        "wordcount",
      ]
    }
    toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
    value="<p>TextEditor</p>"
  >
    <textarea
      id="tiny-react_79065548421612809717342"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </Editor>
</TextEditor>
`;

exports[`TextEditor renders disabled correctly 1`] = `
<TextEditor
  disabled={true}
  init={Object {}}
  inline={false}
  menubar={false}
  onEditorChange={[MockFunction]}
  plugins={
    Array [
      "lists advlist",
      "anchor",
      "autolink",
      "autoresize",
      "charmap",
      "code codesample",
      "directionality",
      "emoticons",
      "fullscreen",
      "help",
      "hr",
      "image imagetools",
      "insertdatetime",
      "link",
      "nonbreaking",
      "paste",
      "preview",
      "print",
      "quickbars",
      "searchreplace",
      "table",
      "toc",
      "visualblocks",
      "wordcount",
    ]
  }
  toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
  value="<p>TextEditor</p>"
>
  <Editor
    apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
    cloudChannel="5"
    disabled={true}
    init={
      Object {
        "menubar": false,
        "selector": undefined,
        "setup": undefined,
      }
    }
    inline={false}
    menubar={false}
    onEditorChange={[MockFunction]}
    plugins={
      Array [
        "lists advlist",
        "anchor",
        "autolink",
        "autoresize",
        "charmap",
        "code codesample",
        "directionality",
        "emoticons",
        "fullscreen",
        "help",
        "hr",
        "image imagetools",
        "insertdatetime",
        "link",
        "nonbreaking",
        "paste",
        "preview",
        "print",
        "quickbars",
        "searchreplace",
        "table",
        "toc",
        "visualblocks",
        "wordcount",
      ]
    }
    toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
    value="<p>TextEditor</p>"
  >
    <textarea
      id="tiny-react_130432841612809717359"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </Editor>
</TextEditor>
`;

exports[`TextEditor renders inline correctly 1`] = `
<TextEditor
  disabled={false}
  init={Object {}}
  inline={true}
  menubar={false}
  onEditorChange={[MockFunction]}
  plugins={
    Array [
      "lists advlist",
      "anchor",
      "autolink",
      "autoresize",
      "charmap",
      "code codesample",
      "directionality",
      "emoticons",
      "fullscreen",
      "help",
      "hr",
      "image imagetools",
      "insertdatetime",
      "link",
      "nonbreaking",
      "paste",
      "preview",
      "print",
      "quickbars",
      "searchreplace",
      "table",
      "toc",
      "visualblocks",
      "wordcount",
    ]
  }
  toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
  value="<p>TextEditor</p>"
>
  <Editor
    apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
    cloudChannel="5"
    disabled={false}
    init={
      Object {
        "menubar": false,
        "selector": undefined,
        "setup": undefined,
      }
    }
    inline={true}
    menubar={false}
    onEditorChange={[MockFunction]}
    plugins={
      Array [
        "lists advlist",
        "anchor",
        "autolink",
        "autoresize",
        "charmap",
        "code codesample",
        "directionality",
        "emoticons",
        "fullscreen",
        "help",
        "hr",
        "image imagetools",
        "insertdatetime",
        "link",
        "nonbreaking",
        "paste",
        "preview",
        "print",
        "quickbars",
        "searchreplace",
        "table",
        "toc",
        "visualblocks",
        "wordcount",
      ]
    }
    toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
    value="<p>TextEditor</p>"
  >
    <div
      id="tiny-react_29179180131612809717357"
    />
  </Editor>
</TextEditor>
`;

exports[`TextEditor renders with menubar correctly 1`] = `
<TextEditor
  disabled={false}
  init={Object {}}
  inline={false}
  menubar={true}
  onEditorChange={[MockFunction]}
  plugins={
    Array [
      "lists advlist",
      "anchor",
      "autolink",
      "autoresize",
      "charmap",
      "code codesample",
      "directionality",
      "emoticons",
      "fullscreen",
      "help",
      "hr",
      "image imagetools",
      "insertdatetime",
      "link",
      "nonbreaking",
      "paste",
      "preview",
      "print",
      "quickbars",
      "searchreplace",
      "table",
      "toc",
      "visualblocks",
      "wordcount",
    ]
  }
  toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
  value="<p>TextEditor</p>"
>
  <Editor
    apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
    cloudChannel="5"
    disabled={false}
    init={
      Object {
        "menubar": true,
        "selector": undefined,
        "setup": undefined,
      }
    }
    inline={false}
    menubar={true}
    onEditorChange={[MockFunction]}
    plugins={
      Array [
        "lists advlist",
        "anchor",
        "autolink",
        "autoresize",
        "charmap",
        "code codesample",
        "directionality",
        "emoticons",
        "fullscreen",
        "help",
        "hr",
        "image imagetools",
        "insertdatetime",
        "link",
        "nonbreaking",
        "paste",
        "preview",
        "print",
        "quickbars",
        "searchreplace",
        "table",
        "toc",
        "visualblocks",
        "wordcount",
      ]
    }
    toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
    value="<p>TextEditor</p>"
  >
    <textarea
      id="tiny-react_40041711051612809717362"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </Editor>
</TextEditor>
`;
